rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -1578,46 +1578,73 @@ void main_window::DecryptSPRXLibraries()
 
 	m_gui_settings->SetValue(gui::fd_decrypt_sprx, QFileInfo(modules.first()).path());
 
-	std::vector<std::string> vec_modules;
+	const auto vec_modules = std::make_shared<std::vector<std::string>>();
 	for (const QString& mod : modules)
 	{
-		vec_modules.push_back(mod.toStdString());
+		vec_modules->push_back(mod.toStdString());
 	}
 
-	const auto input_cb = [this](std::string old_path, std::string path, bool tried) -> std::string
+	auto iterate = std::make_shared<std::function<void(usz, usz)>>();
+	const auto klics_storage = std::make_shared<std::shared_ptr<void>>(); // Opaque storage for function data
+
+	*iterate = [this, iterate, vec_modules, klics_storage](usz mod_index, usz repeat_count)
 	{
-		const QString hint = tr("Hint: KLIC (KLicense key) is a 16-byte long string. (32 hexadecimal characters)"
-			"\nAnd is logged with some sceNpDrm* functions when the game/application which owns \"%0\" is running.").arg(qstr(path));
+		const std::string& path = (*vec_modules)[mod_index];
+		const std::string filename = path.substr(path.find_last_of(fs::delim) + 1);
+
+		const QString hint = tr("Hint: KLIC (KLicense key) is a 16-byte long string. (32 hexadecimal characters, can be prefixed with \"KLIC=0x\" from the log message)"
+			"\nAnd is logged with some sceNpDrm* functions when the game/application which owns \"%0\" is running.").arg(qstr(filename));
 
-		if (tried)
+		if (repeat_count >= 2)
 		{
-			gui_log.error("Failed to decrypt %s with specfied KLIC, retrying.\n%s", old_path, sstr(hint));
+			gui_log.error("Failed to decrypt %s with specfied KLIC, retrying.\n%s", path, sstr(hint));
 		}
 
-		input_dialog dlg(32, "", tr("Enter KLIC of %0").arg(qstr(path)),
-			tried ? tr("Decryption failed with provided KLIC.\n%0").arg(hint) : tr("Hexadecimal only."), "00000000000000000000000000000000", this);
+		input_dialog* dlg = new input_dialog(39, "", tr("Enter KLIC of %0").arg(qstr(filename)),
+			repeat_count >= 2 ? tr("Decryption failed with provided KLIC.\n%0").arg(hint) : tr("Hexadecimal value."), "KLIC=0x00000000000000000000000000000000", this);
 
 		QFont mono = QFontDatabase::systemFont(QFontDatabase::FixedFont);
 		mono.setPointSize(8);
-		dlg.set_input_font(mono, true, '0');
-		dlg.set_clear_button_enabled(false);
-		dlg.set_button_enabled(QDialogButtonBox::StandardButton::Ok, false);
-		dlg.set_validator(new QRegularExpressionValidator(QRegularExpression("^[a-fA-F0-9]*$"))); // HEX only
+		dlg->set_input_font(mono, true, '0');
+		dlg->set_clear_button_enabled(false);
+		dlg->set_button_enabled(QDialogButtonBox::StandardButton::Ok, false);
+		dlg->set_validator(new QRegularExpressionValidator(QRegularExpression("^(K(L(I(C(=(0x?)?)?)?)?)?)?[a-fA-F0-9]{0,32}$"))); // HEX only (with additional klic=0x prefix for convenience))
```

Comment:
This allows uncomplete strings with weird prefix like:
K12345678123456781234567812345678
KL12345678123456781234567812345678
KLI12345678123456781234567812345678
KLIC12345678123456781234567812345678
KLIC=12345678123456781234567812345678
KLIC=012345678123456781234567812345678

If you want the entire part "KLIC=0x" to be optional, you need:
```suggestion
		dlg->set_validator(new QRegularExpressionValidator(QRegularExpression("^(KLIC=0x)?[a-fA-F0-9]{0,32}$"))); // HEX only (with additional klic=0x prefix for convenience))
```

Reply:
No it allows
12345678123456781234567812345678
x12345678123456781234567812345678
0xL12345678123456781234567812345678
=0x12345678123456781234567812345678
C=0x12345678123456781234567812345678
IC=0x12345678123456781234567812345678
LIC=0x12345678123456781234567812345678
KLIC=0x12345678123456781234567812345678

In case someone copies the string from log not in its entirety.

Reply:
It's still wrong though:
E QDEBUG: 12345 (has match)
E QDEBUG: KLIC=0x12345 (has match)
E QDEBUG: **IC=0x12345 (no match)**
E QDEBUG: **KLI12345 (has match)**

What you want is:
```suggestion
		dlg->set_validator(new QRegularExpressionValidator(QRegularExpression("^(((((K?L)?I)?C)?=)?0x)?[a-fA-F0-9]{0,32}$"))); // HEX only (with additional klic=0x prefix for convenience))
```

E QDEBUG: 12345 (has match)
E QDEBUG: KLIC=0x12345 (has match)
E QDEBUG: **IC=0x12345 (has match)**
E QDEBUG: **KLI12345 (no match)**

Reply:
Right, fixed.